pkg/commands/run.go

```diff
@@ -244,8 +244,14 @@ func (c *runCommand) execute(_ *cobra.Command, args []string) {
 		}
 	}()
 
-	ctx, cancel := context.WithTimeout(context.Background(), c.cfg.Run.Timeout)
-	defer cancel()
+	var ctx context.Context
+	if c.cfg.Run.Timeout > 0 {
+		var cancel context.CancelFunc
+		ctx, cancel = context.WithTimeout(context.Background(), c.cfg.Run.Timeout)
+		defer cancel()
+	} else {
+		ctx = context.Background()
+	}
```

Comment:
I would have written this

```suggestion
	ctx := context.Background()
	if c.cfg.Run.Timeout > 0 {
		var cancel context.CancelFunc
		ctx, cancel = context.WithTimeout(ctx, c.cfg.Run.Timeout)
		defer cancel()
	}
```

But I'm surprised because if the scope of cancel 🤔

Shouldn't be this?

```suggestion
	ctx := context.Background()
	var cancel context.CancelFunc
	if c.cfg.Run.Timeout > 0 {
		ctx, cancel = context.WithTimeout(ctx, c.cfg.Run.Timeout)
		defer cancel()
	}
```

Reply:
Scope isn't relevant here, you can call `defer` in any path/scope and it will execute when the function returns so this is fine.